pkg/resource/hosted_zone/hook.go

```diff
@@ -0,0 +1,15 @@
+package hosted_zone
+
+import (
+	"fmt"
+	"time"
+
+	svcapitypes "github.com/aws-controllers-k8s/route53-controller/apis/v1alpha1"
+)
+
+// getCallerReference will generate a CallerReference for a given hosted zone
+// using the name of the zone and the current timestamp, so that it produces a
+// unique value
+func getCallerReference(zone *svcapitypes.HostedZone) string {
+	return fmt.Sprintf("%s-%d", *zone.Spec.Name, time.Now().UnixMilli())
+}
```

Comment:
Not sure I am a fan of the name of this field (or having it manually set by the user vs. the SDK setting it on a session-level), but meh, this is the reasoning behind this field:

https://aws.amazon.com/builders-library/making-retries-safe-with-idempotent-APIs/